Added Indent code functionality with Tab key and Unit tested the Tab key

jayrexacilo opened the pull request:
- Added two spaces "\xa0\xa0" when Indenting the code on Tab key
- Resolve issue #15

ajayrexluega committed:
Added Indent with Tab key functionality and Unit tested the Tab key

Matthew-Burfield reviewed (changes requested):
Yeah I'm happy with the double space solution, it's better than having inconsistency in the spacing. We can potentially do a spaces vs tabs setting later on.

Just one thing with the tests, each test should only test one thing. We already have tests that test "abc123" and enter etc. So in your test, could you update it so it only triggers the tab key?

jayrexacilo commented:
.text() api removes extra spaces. so when i test it, it'll fail. see image

![abc-tab-test](https://user-images.githubusercontent.com/31985700/37712348-96de917a-2d4e-11e8-8f55-7d9d82977512.png)
![tab-test](https://user-images.githubusercontent.com/31985700/37712349-9795890c-2d4e-11e8-8cca-daa71e352d5f.png)

Matthew-Burfield committed:
Update Editor.spec.js

You're totally right that .text() trims strings, I'm not sure the best way to test this then - potentially we could use the .html() and check that it contains two spaces?

Otherwise, I've added a test that checks the documentData object directly (not usually good practice as it's testing the implementation details of the Vue component as opposed to the end result the user sees)

Anyway, what do you think?

Matthew-Burfield commented:
You're totally right @jayrexacilo @ajayrexluega 

You're totally right that .text() trims strings, I'm not sure the best way to test this then - potentially we could use the .html() and check that it contains two spaces?

Otherwise, I've added a test that checks the documentData object directly (not usually good practice as it's testing the implementation details of the Vue component as opposed to the end result the user sees)

Anyway, what do you think?
